lib/project_types/theme/models/specification_handlers/theme.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Theme
+  module Models
+    module SpecificationHandlers
+      class Theme
+        REQUIRED_FOLDERS = %w(config layout sections templates).map { |folder| "#{folder}/" }
+
+        def initialize(root)
+          Dir.chdir(root) do
+            self.folders = Dir["*/"] + Dir["templates/*/"]
+          end
+        end
+
+        def valid?
+          validate
+          missing_folders.empty?
+        end
+
+        private
+
+        attr_accessor :folders
+        attr_accessor :missing_folders
+
+        def validate
```

Comment:
How many directories do templates have? If it's not too many, I'd refrain from having an internal state, `missing_folders`, that's internally mutated as a side effect of doing a validation. What about?

```rb
def missing_folders
  REQUIRED_FOLDERS - folders
end
```

Reply:
I was inspired by Rails (calling `.valid?` to get the correct results for `.errors`) for this solution, but I agree, so I changed it :)